fix(stdune): make sure Path.drop_prefix drops path prefixes only

Alizter opened the pull request:
Fixes #8870

Alizter committed:
fix(stdune): make sure Path.drop_prefix drops path prefixes only

Signed-off-by: Ali Caglayan <alizter@gmail.com>

emillon commented:
FYI the issue is affecting our own test suite since in the describe tests, when replacing `lib/ocaml` to `FINDLIB`, we're replacing `lib/ocaml-compiler-libs` by `FINDLIB/-compiler-libs`. This happens with OCaml 5.1 which might have a slightly different dir layout.

Alizter commented:
@rgrinberg Why was this marked as draft?

rgrinberg commented:
b/c the tests are failing

Alizter commented:
Sorry @rgrinberg you are correct that the failures are relevant. I had a look and I couldn't work out how to fix melange here. I might have a look again at a later date.

Alizter commented:
@rgrinberg @anmonteiro Melange is now fixed. I was missing the case where we drop the entire path which should give just `Local.root`. I've updated the tests to check this case too.

Alizter commented:
@rgrinberg I've created https://github.com/ocaml/dune/pull/9363 for the extra tests I added here.

rgrinberg commented on otherlibs/stdune/test/path_tests.ml:
```diff
@@ -477,20 +477,34 @@ let%expect_test "drop external prefix" =
+  [%expect {| None |}]
+;;
+
+let%expect_test "drop entire path" =
```
Tests are still being added in this PR

Alizter replied:
Done.

Alizter commented:
@rgrinberg Sorry about those extra tests, not sure what happened with git but it resolved the conflict in a funny way causing the tests to be duplicated. Should all be fixed now.
